[MTS][87766816] Compile .js to dist/ directory

michaelstalker opened the pull request:
https://www.pivotaltracker.com/story/show/87766816
https://www.pivotaltracker.com/story/show/88072428

This will make the library more consistent with other JavaScript libraries at the company.

michaelstalker committed:
[MTS][87766816] Move .js to dist/

I also added some scripts to package.json. These changes will make
the library more consistent with our other JavaScript libraries.

warrenv commented on gulpfile.js:
```diff
@@ -9,13 +9,13 @@ var del = require('del');
+  .pipe(gulp.dest('./dist'));
 });
 
 gulp.task('coffee', function() {
```
Other than an argument difference to coffee() - this task is the same as diff. Could one be eliminated?

michaelstalker replied:
Good catch. Yes, let's delete one.
1. Do we want the `bare` option set to `true`? Your [templates](https://github.com/rentpath/generator-rentpath-flight/blob/dev/app/templates/gulpfile.js) have that set to `true`, which compiles the scripts without a top-level function wrapper.
2. Do we want to use the `header` option, which prepends `// Generated by CoffeeScript 1.9.0` to the top of each file. This is useful for development, but not so useful in QA and production. Maybe AG's build process strips out these comments in production. Do you know if it does?

c0 replied:
Comments are stripped out.

warrenv commented:
Please update the README to include instructions for running 'npm install' and 'bower install'.

warrenv commented:
The tests are not running for me:

it-mbp-wvosper:map.js wvosper$ karma start
INFO [karma]: Karma v0.12.31 server started at http://localhost:9876/
INFO [launcher]: Starting browser Chrome
WARN [watcher]: Pattern "/Users/wvosper/source/map.js/app/bower_components/jquery/dist/jquery.js" does not match any file.
INFO [Chrome 40.0.2214 (Mac OS X 10.10.2)]: Connected on socket FDI5ehmuFU8KmDr8rG0M with id 95695259
Chrome 40.0.2214 (Mac OS X 10.10.2) ERROR: 'There is no timestamp for /base/app/bower_components/jquery/jquery.js!'

WARN [web-server]: 404: /base/app/bower_components/jquery/jquery.js
Chrome 40.0.2214 (Mac OS X 10.10.2) ERROR
  Uncaught Error: Script error for: jquery
  http://requirejs.org/docs/errors.html#scripterror
  at /Users/wvosper/source/map.js/node_modules/requirejs/require.js:141

warrenv commented:
What do you think about adding the 'scripts' block from package.json in gallery.js?

michaelstalker commented:
@warrenv I updated the README, fixed the test, and added the `scripts` block from `package.json` in gallery.js. I removed some scripts that weren't relevant map.js, like deleting the `.tmp/` directory. I included much of the README information from gallery.js.

warrenv commented:
:+1:  Very nice work - removing the need to compile coffeescript to .tmp is a nice feature that we can probably adopt for the other js repos.
